lib/internal/repl.js

```diff
@@ -164,13 +164,22 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
       }
     }
 
-    fs.open(historyPath, 'w', onhandle);
+    fs.open(historyPath, 'r+', onhandle);
```

Comment:
Won't this break creating a new file if none exists? [[1]](https://nodejs.org/dist/latest-v7.x/docs/api/fs.html#fs_fs_open_path_flags_mode_callback)